### DISCUSSION TITLE
Address as Field Type

### DISCUSSION THREAD

**Discussion by RitterKnightCreative** [craftcms/cms#11438] [Ideas] [answered]
I can think of a bunch of scenarios where you might want to attach an Address to an entry, asset or category and be able to query them easily — eg show me all the stores or corporate locations in a certain state or a list of venues that a certain band playing.

Are there any plans to extend the Address element to be a field type that can be attached to any element — not just users? 

One of things that makes Craft great is it makes no assumptions about your content... except in this case, Address is somewhat limited, aside from e-commerce.  Hopefully that will change.

**Comment by bossanova808** [comment]
This would be great - a standardised address field type would be so useful (ideally with an option to be powered by Google address search, and a fall back to manual entry....)

**Comment by brandonkelly** [comment]
We had hoped to get this into Craft 4, but ran into some technical challenges and had to postpone it. Definitely still planned though.

**Reply by digason** [reply]
Is there a planned milestone for allowing addresses on elements other than users?

**Comment by BenParizek** [comment]
For folks who may have used Sprout Fields Address Field in the past, we will not be updating it to Craft 4 as Craft Commerce now has an Address Field using the same underlying library and a native, and Craft adding a first party solution to this will remove the need for the third-party field type.

**Comment by jrrdnx** [comment]
Hoping for this update as well. Converting a current client over from an older ExpressionEngine build, and would love to be able to use an Address field for their multiple office locations around the world.

**Comment by alexanderbuergin** [comment]
I would love to see that feature

**Comment by brandonkelly** [comment, marked as answer]
Craft 5 added a new Addresses field type 🚀

**Reply by alexanderloewe** [reply]
Cool
